lib/trento/application/integration/discovery/protocol/enrich_request_host_deregistration.ex

```diff
@@ -17,32 +22,41 @@ defimpl Trento.Support.Middleware.Enrichable,
           | {:error, :host_alive}
           | {:error, :host_not_registered}
   def enrich(%RequestHostDeregistration{host_id: host_id} = command, _) do
-    Repo.get(HostReadModel, host_id)
-    |> Repo.preload([:heartbeat_timestamp])
+    HostReadModel
+    |> where([h], h.id == ^host_id)
+    |> enrich_host_read_model_query()
+    |> Repo.one()
     |> host_deregisterable(command)
   end
 
   defp host_deregisterable(
-         %HostReadModel{heartbeat_timestamp: nil, deregistered_at: nil},
+         %HostReadModel{last_heartbeat_timestamp: nil, deregistered_at: nil},
          %RequestHostDeregistration{} = command
        ),
        do: {:ok, command}
 
   defp host_deregisterable(
          %HostReadModel{
-           heartbeat_timestamp: %Trento.Heartbeat{timestamp: timestamp},
+           last_heartbeat_timestamp: last_heartbeat_timestamp,
            deregistered_at: nil
          },
          %RequestHostDeregistration{} = command
        ) do
     if :lt ==
          DateTime.compare(
            DateTime.utc_now(),
-           DateTime.add(timestamp, @total_deregistration_debounce, :millisecond)
+           DateTime.add(last_heartbeat_timestamp, @total_deregistration_debounce, :millisecond)
          ),
        do: {:error, :host_alive},
        else: {:ok, command}
   end
 
   defp host_deregisterable(_, _), do: {:error, :host_not_registered}
+
+  @spec enrich_host_read_model_query(Ecto.Query.t()) :: Ecto.Query.t()
```

Comment:
Why don't you add a `get_host_by_id` or similar in the usecases file?
You would reuse the enrichment function there, and here you could use this instead of running the query here.

Reply:
It's just query related, I think we could safely place it in the Readmodel itself and call it from other parts

So you can pass enrich readmodel related query as you need